homeassistant/components/n26/sensor.py

```diff
@@ -0,0 +1,312 @@
+        """Initialize a N26 card."""
+        self._hass = hass
+        self._client = client
+        self._account_name = account_name
+        self._card = card
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True.
+
+        Data needs to be polled from the bank servers.
+        """
+        return True
+
+    def update(self) -> None:
+        """Get the current balance and currency for the account."""
+        from .const import LAST_CARDS_UPDATE
+        if (datetime.now() < self._hass.data[DOMAIN][LAST_CARDS_UPDATE] +
+                (SCAN_INTERVAL / 2)):
+            return
+
+        self._hass.data[DOMAIN][CARDS] = self._client.get_cards()
+        self._hass.data[DOMAIN][LAST_CARDS_UPDATE] = datetime.now()
+
+        for card in self._hass.data[DOMAIN][CARDS]:
+            if card["id"] == self._card["id"]:
+                self._card = card
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the entity."""
+        return self._card["id"]
+
+    @property
+    def name(self) -> str:
+        """Friendly name of the sensor."""
+        return "%s_card_%s" % (self._account_name.lower(), self._card["id"])
+
+    @property
+    def state(self) -> float:
+        """Return the balance of the account as state."""
+        return self._card["status"]
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Use the currency as unit of measurement."""
+        return ""
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Additional attributes of the sensor."""
+        attributes = {
+            "apple_pay_eligible": self._card.get("applePayEligible"),
+            "card_activated": timestamp_ms_to_date(
+                self._card.get("cardActivated")),
+            "card_product": self._card.get("cardProduct"),
+            "card_product_type": self._card.get("cardProductType"),
+            "card_settings_id": self._card.get("cardSettingsId"),
+            "card_Type": self._card.get("cardType"),
+            "design": self._card.get("design"),
+            "exceet_actual_delivery_date":
+                self._card.get("exceetActualDeliveryDate"),
+            "exceet_card_status": self._card.get("exceetCardStatus"),
+            "exceet_expected_delivery_date":
+                self._card.get("exceetExpectedDeliveryDate"),
+            "exceet_express_card_delivery":
+                self._card.get("exceetExpressCardDelivery"),
+            "exceet_express_card_delivery_email_sent":
+                self._card.get("exceetExpressCardDeliveryEmailSent"),
+            "exceet_express_card_delivery_tracking_id":
+                self._card.get("exceetExpressCardDeliveryTrackingId"),
+            "expiration_date": timestamp_ms_to_date(
+                self._card.get("expirationDate")),
+            "google_pay_eligible": self._card.get("googlePayEligible"),
+            "id": self._card.get("id"),
```

Comment:
We try to not include static information like ids in the state attributes. The state attributes are for info related to the state of the entity.

This is very many attributes for one entity. I'd recommend reviewing them and removing the ones that won't change during entity lifetime.

Reply:
Well... some of these attributes will probably change a single time (like `pin_defined`) but all of them contain contain information I would like to have available in hass (and db which is used elsewhere).
I removed `order_id` and `public_token` (although this one will change it is not of any use in hass without using the n26 library or some other http client directly).